src/js/modules/infragistics.ui.widget.js

```diff
@@ -98,36 +99,93 @@
 				return $.ig.regional.defaults[ key ];
 			}
 		},
+		_getLocaleWidgetName: function () {
+			this.localeWidgetName = this.localeWidgetName ||
+							(this.widgetFullName || this.widgetName)
+								.replace(/^((ui-ig(\.)*)|(ig(\.)*))/ig, "");
+			return this.localeWidgetName;
+		},
+		_getLocaleObject: function (locale) {
+			return $.ig.locale &&
+					$.ig.locale[ locale ] &&
+					$.ig.locale[ locale ][ this._getLocaleWidgetName() ];
+		},
 		_getLocaleValue: function (key) {
-			var locale = this.options.locale,
-				widgetName = this.localeWidgetName || this.widgetName.split("ig")[ 1 ];
-
-			if (locale && locale[ key ]) {
-				return locale[ key ];
-			} else if ($.ig && $.ig[ widgetName ] && $.ig[ widgetName ].locale) {
-				return $.ig[ widgetName ].locale[ key ];
-			} else {
-				return "";
+			var locale = this.options.locale, localeObject, localeWidgetName;
```

Comment:
Why do you need a `_getLocaleObject` method, if further locale object processing is done in other methods calling it?